modules/apps/object/object-dynamic-data-mapping/src/main/java/com/liferay/object/dynamic/data/mapping/internal/storage/ObjectDDMStorageAdapter.java

```diff
@@ -153,51 +154,77 @@ public DDMStorageAdapterSaveResponse save(
+	private Value _getDDMFormFieldValue(
+		Map<String, DDMFormField> ddmFormFieldsMap, Locale defaultLocale,
+		Map<String, Object> properties, DDMFormField ddmFormField) {
 
-		DDMFormValues ddmFormValues = new DDMFormValues(ddmForm);
+		Value value = new LocalizedValue(defaultLocale);
 
-		ddmFormValues.addAvailableLocale(ddmForm.getDefaultLocale());
+		Object objectFieldValue = properties.get(
+			_getObjectFieldName(ddmFormFieldsMap.get(ddmFormField.getName())));
 
-		Map<String, DDMFormField> ddmFormFieldsMap =
-			ddmForm.getDDMFormFieldsMap(true);
-		Map<String, Object> properties = objectEntry.getProperties();
+		if (objectFieldValue instanceof Double) {
+			NumberFormat numberFormat = NumberFormat.getInstance(defaultLocale);
 
-		for (DDMFormField ddmFormField : ddmForm.getDDMFormFields()) {
-			DDMFormFieldValue ddmFormFieldValue = new DDMFormFieldValue();
+			value.addString(
+				defaultLocale, numberFormat.format(objectFieldValue));
+		}
+		else if (objectFieldValue instanceof byte[]) {
+			value.addString(
+				defaultLocale, new String((byte[])objectFieldValue));
+		}
+		else {
+			value.addString(defaultLocale, String.valueOf(objectFieldValue));
+		}
 
-			ddmFormFieldValue.setName(ddmFormField.getName());
+		return value;
+	}
 
-			Value value = new LocalizedValue(ddmForm.getDefaultLocale());
+	private List<DDMFormFieldValue> _getDDMFormFieldValues(
+		List<DDMFormField> ddmFormFields,
+		Map<String, DDMFormField> ddmFormFieldsMap, Locale defaultLocale,
+		Map<String, Object> properties) {
 
-			Object objectFieldValue = properties.get(
-				_getObjectFieldName(
-					ddmFormFieldsMap.get(ddmFormField.getName())));
+		List<DDMFormFieldValue> ddmFormFieldValues = new ArrayList<>();
 
-			if (objectFieldValue instanceof Double) {
-				NumberFormat numberFormat = NumberFormat.getInstance(
-					ddmForm.getDefaultLocale());
+		ddmFormFields.forEach(
+			ddmFormField -> {
+				if (StringUtil.equals(
+						ddmFormField.getType(),
+						DDMFormFieldTypeConstants.FIELDSET)) {
 
-				value.addString(
-					ddmForm.getDefaultLocale(),
-					numberFormat.format(objectFieldValue));
-			}
-			else if (objectFieldValue instanceof byte[]) {
-				value.addString(
-					ddmForm.getDefaultLocale(),
-					new String((byte[])objectFieldValue));
-			}
-			else {
-				value.addString(
-					ddmForm.getDefaultLocale(),
-					String.valueOf(objectFieldValue));
-			}
+					ddmFormFieldValues.addAll(
+						_getDDMFormFieldValues(
+							ddmFormField.getNestedDDMFormFields(),
+							ddmFormFieldsMap, defaultLocale, properties));
+				}
 
-			ddmFormFieldValue.setValue(value);
+				DDMFormFieldValue ddmFormFieldValue = new DDMFormFieldValue();
 
-			ddmFormValues.addDDMFormFieldValue(ddmFormFieldValue);
-		}
+				ddmFormFieldValue.setName(ddmFormField.getName());
+
+				Value value = _getDDMFormFieldValue(
+					ddmFormFieldsMap, defaultLocale, properties, ddmFormField);
+
+				ddmFormFieldValue.setValue(value);
```

Comment:
@mateussgn can we inline the `value` variable?

Reply:
Sure!
Thanks for the tip!